converter.py

```diff
@@ -0,0 +1,275 @@
+"""
+Convert existing jupyter notebook to an airbnb knowledge repo format
+"""
+import os
+import re
+import json
+import subprocess
+from dateutil import parser as date_parser
+
+
+def main(ml_repo, knowledge_repo, inplace):
+    ml_repo_path = os.path.abspath(ml_repo)
+    knowledge_repo_path = os.path.abspath(knowledge_repo)
+    if not os.path.isdir(knowledge_repo_path):
+        init_knowledge_repo(knowledge_repo_path)
+
+    convert_all_posts(ml_repo_path, knowledge_repo_path, inplace)
+
+
+def init_knowledge_repo(path):
+    cmd = 'knowledge_repo --repo {} init'.format(path)
+    subprocess.call(cmd, shell = True)
+
+
+def convert_all_posts(path, knowledge_repo_path, inplace):
+    """Recursive walk down all directory to perform the conversion"""
+    if os.path.isdir(path):
+        files = [os.path.join(path, f) for f in os.listdir(path)]
+        for f in files:
+            convert_all_posts(f, knowledge_repo_path, inplace)
+    elif '-converted' not in path:
+        head, ext = os.path.splitext(path)
+        if ext == ".ipynb":
+            try:
+                converter = IpynbConverter(knowledge_repo_path, inplace)
+                notebook = converter.convert(path)
+                print(path)
+                print(converter._path)
+                converter.add(notebook)
+            except Exception as e:
+                print('Skipping: {}'.format(path))
+                print(e)
+
+
+class IpynbConverter:
+    """
+    Converts Jupyter notebook to airbnb knowledge repo format [1]_.
+
+    Parameters
+    ----------
+    knowledge_repo_path : str
+        Path to store the airbnb knowledge repo-ed notebook.
+
+    inplace : bool
+        Whether to perform the conversion inplace or not. If
+        false, then it will create a new notebook that has the
+        '-converted' appended to the file name.
+
+    Attributes
+    ----------
+    date_created_ : str
```

Comment:
Why the trailing underscores? Usually we use trailing underscores to avoid conflicts with built in functions (`min_` vs. `min()`)

Reply:
Writing style is heavily influenced by [scikit-learn](http://scikit-learn.org/stable/developers/contributing.html#estimated-attributes).